cmd/metrics-scraper/app/metricsscraper.go

```diff
@@ -0,0 +1,79 @@
+package app
+
+import (
+	"context"
+	"fmt"
+	"github.com/gin-gonic/gin"
+	"github.com/karmada-io/dashboard/cmd/api/app/router"
+	"github.com/karmada-io/dashboard/cmd/web/app/options"
+	"github.com/karmada-io/dashboard/pkg/config"
+	"github.com/karmada-io/dashboard/pkg/environment"
+	"github.com/karmada-io/karmada/pkg/sharedcli/klogflag"
+	"github.com/spf13/cobra"
+	cliflag "k8s.io/component-base/cli/flag"
+	"k8s.io/klog/v2"
+	"net/http"
+	"os"
+)
+
+// NewMetricsScraperCommand creates a *cobra.Command object with default parameters
+func NewMetricsScraperCommand(ctx context.Context) *cobra.Command {
+	opts := options.NewOptions()
+	cmd := &cobra.Command{
+		Use:  "karmada-dashboard-metrics-scraper",
+		Long: `The karmada-dashboard-metrics-scraper responsible for scraping and visualizing the metrics of karmada components. `,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if err := run(ctx, opts); err != nil {
+				return err
+			}
+			return nil
+		},
+		Args: func(cmd *cobra.Command, args []string) error {
+			for _, arg := range args {
+				if len(arg) > 0 {
+					return fmt.Errorf("%q does not take any arguments, got %q", cmd.CommandPath(), args)
+				}
+			}
+			return nil
+		},
+	}
+	fss := cliflag.NamedFlagSets{}
+
+	genericFlagSet := fss.FlagSet("generic")
+	opts.AddFlags(genericFlagSet)
+
+	// Set klog flags
+	logsFlagSet := fss.FlagSet("logs")
+	klogflag.Add(logsFlagSet)
+
+	cmd.Flags().AddFlagSet(genericFlagSet)
+	cmd.Flags().AddFlagSet(logsFlagSet)
+	return cmd
+}
+
+func run(ctx context.Context, opts *options.Options) error {
+	klog.InfoS("Starting Karmada Dashboard Metrics-scraper", "version", environment.Version)
+	serve(opts)
+	select {
+	case <-ctx.Done():
+		os.Exit(0)
+	}
+	return nil
+}
+
+func serve(opts *options.Options) {
+	insecureAddress := fmt.Sprintf("%s:%d", opts.InsecureBindAddress, opts.InsecurePort)
```

Comment:
here, it seems that the you reused the option for metrics-scrape, two problems:
- in the metrics-scape module, we need use client with client-go sdk, we should init the client here.
- the options should be seperated, another copy of options maybe a good way.